Sources/PointFree/Episode/Show.swift

```diff
@@ -38,7 +38,125 @@ let episodeResponse: Middleware<StatusLineOpen, ResponseEnded, Tuple4<Either<Str
     }
 )
 
-let episodeView = View<(Database.User?, Stripe.Subscription.Status?, Episode)> { user, subscriptionStatus, episode in
+let useCreditResponse =
+  filterMap(
+    over1(episode(forParam:)) >>> require1 >>> pure,
+    or: writeStatus(.notFound) >-> respond(episodeNotFoundView.contramap(lower))
+    )
+    <<< { userEpisodePermission >-> $0 } // Is there a name for this and/or do we want to introduce someting like "lift" to do it?
+    <<< filterMap(require3 >>> pure, or: loginAndRedirect)
+    <<< validateCreditRequest
+    <| applyCreditMiddleware
+
+private func applyCreditMiddleware<Z>(
+  _ conn: Conn<StatusLineOpen, T4<EpisodePermission, Episode, Database.User, Z>>
+  ) -> IO<Conn<ResponseEnded, Data>> {
+
+  let (episode, user) = (get2(conn.data), get3(conn.data))
+
+  guard user.episodeCreditCount > 0 else {
+    return conn
+      |> redirect(
+        to: .episode(.left(episode.slug)),
+        headersMiddleware: flash(.error, "You do not have any credits to use.")
+    )
+  }
+
+  return AppEnvironment.current.database.addEpisodeCredit(episode.sequence, user.id)
```

Comment:
Lol sorry to get bikeshed-y, but "add credit" feels like adding to credit for later redeeming. Maybe this should be `redeemCredit`?

Reply:
yeah def, can do